internal/util.go

```diff
@@ -12,13 +12,17 @@ import (
 // for resources that can contain an arbitrary or dynamic number of fields.
 func RemainingKeys(s interface{}, m map[string]interface{}) (extras map[string]interface{}) {
 	extras = make(map[string]interface{})
+	for k, v := range m {
+		extras[k] = v
+	}
+
 	valueOf := reflect.ValueOf(s)
 	typeOf := reflect.TypeOf(s)
 	for i := 0; i < valueOf.NumField(); i++ {
 		field := typeOf.Field(i)
 		tagValue := field.Tag.Get("json")
-		if _, ok := m[tagValue]; !ok {
-			extras[tagValue] = m[tagValue]
+		if _, ok := extras[tagValue]; ok {
+			delete(extras, tagValue)
```

Comment:
The function was actually wrong to begin with. I just thought it was correct because the unit test said so. It was only after I started implementing the Keystone v3 User support that I noticed it was wrong.

The way the previous function worked was that it was looking a json tag. However, the function is supposed to collect fields that have no json tags. `tagValue` is then `""`.

So I switched the logic around to delete known json fields and return the remaining result.